src/heuristic.c

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (C) 2015  University of Alberta
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/**
+ * @file heuristic.c
+ * @author Scott Chu
+ * @date 2020-05-28
+ */
+
+/*File Includes*/
+#include "../inc/heuristic.h"
+#include "cloud_detection.c"
+
+/*Standard Lib Includes*/
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <math.h>
```

Comment:
Not 100% sure if all of these exist: maybe we won't worry for now, but these are all GNU standard files, and sadly we will not be compiling with GNU, but rather the TI compiler, so these either won't exist, or may be a bit different.